### PR TITLE
OpenAPI spec update for 1889

### PR DESCRIPTION
**Pull request by SferaDev**
Add vector similarity search endpoint to the database repository.

### CODE REVIEW

File: packages/client/src/schema/repository.ts [hunk 1]
```diff
@@ -738,6 +739,35 @@ export abstract class Repository<Record extends XataRecord> extends Query<
     }
   ): Promise<SearchXataRecord<SelectedPick<Record, ['*']>>[]>;
 
+  /**
+   * Search for vectors in the table.
+   * @param column The column to search for.
+   * @param query The vector to search for similarities. Must have the same dimension as the vector column used.
+   * @param options The options to search with (like: spaceFunction)
+   */
+  abstract searchVector<F extends SelectableColumn<Record>>(
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:748]:
```suggestion
  abstract vectorSearch<F extends SelectableColumn<Record>>(
```

File: packages/client/src/schema/repository.ts [hunk 2]
```diff
@@ -1704,6 +1734,44 @@ export class RestRepository<Record extends XataRecord>
     });
   }
 
+  async searchVector<F extends SelectableColumn<Record, []>>(
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:1737]:
```suggestion
  async vectorSearch<F extends SelectableColumn<Record, []>>(
```

File: packages/client/src/schema/repository.ts [hunk 3]
```diff
@@ -37,7 +38,7 @@ import { cleanFilter, Filter } from './filters';
 import { Page } from './pagination';
 import { Query } from './query';
 import { EditableData, Identifiable, isIdentifiable, XataRecord } from './record';
-import { SelectableColumn, SelectedPick } from './selection';
+import { ColumnsByValue, SelectableColumn, SelectedPick, ValueAtColumn } from './selection';
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:41]:
```suggestion
import { ColumnsByValue, SelectableColumn, SelectedPick } from './selection';
```

File: packages/client/src/schema/repository.ts [hunk 4]
```diff
@@ -1704,6 +1734,44 @@ export class RestRepository<Record extends XataRecord>
...
+        pathParams: {
+          workspace: '{workspaceId}',
+          dbBranchName: '{dbBranch}',
+          region: '{region}',
+          tableName: this.#table
+        },
+        body: {
+          column,
+          queryVector: query,
+          spaceFunction: options?.spaceFunction,
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:1761]:
```suggestion
          similarityFunction: options?.similarityFunction,
```

File: test/integration/vectorSearch.test.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  test('with filter', async () => {
+    const results = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4], {
+      filter: { full_name: { $any: ['r3', 'r4'] } }
+    });
+
+    expect(results.map((r) => r.full_name)).toEqual(['r4', 'r3']);
+  });
+
+  test('euclidean', async () => {
+    const results = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4], { spaceFunction: 'l1' });
```
Comment (SferaDev) [inline test/integration/vectorSearch.test.ts:68]:
```suggestion
    const results = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4], { similarityFunction: 'l1' });
```

File: test/integration/vectorSearch.test.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    });
+
+    expect(results.map((r) => r.full_name)).toEqual(['r4']);
+  });
+
+  test('with filter and size and spaceFunction', async () => {
+    const results = await xata.db.users.vectorSearch('vector', [1, 2, 3, 4], {
+      filter: { full_name: { $any: ['r3', 'r4'] } },
+      size: 1,
+      spaceFunction: 'l1'
```
Comment (SferaDev) [inline test/integration/vectorSearch.test.ts:92]:
```suggestion
      similarityFunction: 'l1'
```

File: packages/client/src/schema/repository.ts [hunk 7]
```diff
@@ -1704,6 +1734,44 @@ export class RestRepository<Record extends XataRecord>
     });
   }
 
+  async vectorSearch<F extends ColumnsByValue<Record, number[]>>(
+    column: F,
+    query: number[],
+    options?:
+      | {
+          spaceFunction?: string | undefined;
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:1742]:
```suggestion
          similarityFunction?: string | undefined;
```

**Review by richardgill — APPROVED** [review]
LGTM

**Comment by SferaDev** [comment]
Disabled E2E tests until https://github.com/xataio/xata/pull/1940 is merged and deployed. It's a small issue I found on the new endpoint. Merging this to be able to deploy the frontend.

**Pull request merged**